src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java

```diff
@@ -1178,11 +1187,15 @@ public static boolean isInstance(final Object value, final Type type) {
      * @return a non-empty array containing the upper bounds of the wildcard
      * type.
      */
-    public static Type[] getImplicitUpperBounds(final WildcardType wildcardType) {
-        Validate.notNull(wildcardType, "wildcardType is null");
-        final Type[] bounds = wildcardType.getUpperBounds();
-
-        return bounds.length == 0 ? new Type[] { Object.class } : normalizeUpperBounds(bounds);
+    @SuppressWarnings({"assignment.type.incompatible","return.type.incompatible"}) /*
+    #1 Validate.notNull() => wildcardType.getImplicitUpperBounds() returns an array of @MinLen(1)
+    #2 new Type[] {Object.class} has length 1
+    */
+    public static Type @MinLen(1) [] getImplicitUpperBounds(final WildcardType wildcardType) {
+        Validate.notNull(wildcardType, "wildcardType is null"); // #1
+        final Type @MinLen(1) [] bounds = wildcardType.getUpperBounds();
```

Comment:
It looks like the `bounds` local variable can be an empty array and therefore should not be annotated `@MinLen(1)`.